src/AdoPat/PatClient.cs

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AdoPat
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.VisualStudio.Services.DelegatedAuthorization;
+
+    /// <summary>
+    /// An abstraction over common operations with the Azure DevOps PAT Lifecycle Management REST API.
+    /// </summary>
+    public class PatClient
+    {
+        private const int PageSize = 100; // Using the maximum allowable page size allows us to reduce HTTP calls.
+        private const bool AllOrgs = false; // Azure DevOps have recommended we always set this to false.
+
+        private ITokensHttpClientWrapper client;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PatClient"/> class.
+        /// </summary>
+        /// <param name="client">Any class which implements the <see cref="ITokensHttpClientWrapper"/> interface.</param>
+        public PatClient(ITokensHttpClientWrapper client)
+        {
+            this.client = client;
+        }
+
+        /// <summary>
+        /// Creates a new PAT.
+        /// </summary>
+        /// <param name="patTokenCreateRequest">
+        /// XXX: Not documented by Visual Studio client libraries. Should we depend on this or
+        /// should this method use the corresponding primitives as arguments? Not yet clear.
+        /// </param>
+        /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
+        /// <returns>A <see cref="Task"/> returning a <see cref="PatTokenResult"/>.</returns>
+        public async Task<PatToken> CreatePatAsync(
+            PatTokenCreateRequest patTokenCreateRequest,
+            CancellationToken cancellationToken = default)
+        {
+            var patTokenResult = await this.client.CreatePatAsync(
+                patTokenCreateRequest,
+                cancellationToken: cancellationToken)
+            .ConfigureAwait(false);
+
+            if (patTokenResult.PatToken == null)
+            {
+                throw new PatClientException($"Failed to create PAT during regeneration: {patTokenResult.PatTokenError}");

```

Comment:
Shall we handle first time PAT creation message differently?

Reply:
Uhh... probably we should just leave out the `during regeneration` part here. In regeneration or otherwise it's still just creating a PAT. I think `Failed to create PAT` will be enough.